packages/strapi-design-system/src/DatePicker/DatePicker.js

```diff
@@ -0,0 +1,124 @@
+import React, { useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import CalendarIcon from '@strapi/icons/Calendar';
+import styled from 'styled-components';
+import { Popover } from '../Popover';
+import { Table, Thead, Tbody, Tr } from '../Table';
+import { DatePickerTh } from './DatePickerTh';
+import { DatePickerTd } from './DatePickerTd';
+import { generateWeeks } from './generateWeeks';
+import { FocusTrap } from '../FocusTrap';
+import { TextInput } from '../TextInput';
+import { VisuallyHidden } from '../VisuallyHidden';
+
+const DatePickerButton = styled.button`
+  border: none;
+  background: transparent;
+  border-radius: ${({ theme }) => theme.borderRadius};
+`;
+
+export const DatePicker = ({ initialDate, selectedDate, onChange, label, selectedDateLabel, ...props }) => {
+  const [visible, setVisible] = useState(false);
+  const inputRef = useRef(null);
+
+  const [weeks, activeRow, activeCol] = generateWeeks(initialDate, selectedDate);
+
+  const langFormatter = new Intl.DateTimeFormat();
+
+  const formattedDate = selectedDate ? langFormatter.format(selectedDate) : '';
+  const placeholder = langFormatter.format(new Date(1970, 0, 1));
+
+  const handleSelectDay = (date) => {
+    onChange(date);
+    setVisible(false);
+  };
+
+  return (
+    <div>
+      <TextInput
+        ref={inputRef}
+        onClick={() => setVisible((s) => !s)}
+        // Prevent input from changing for now
+        onChange={() => {}}
+        value={formattedDate}
+        placeholder={placeholder}
+        leftAction={
+          <DatePickerButton
+            onClick={() => setVisible((s) => !s)}
+            aria-label={selectedDate ? selectedDateLabel(langFormatter.format(selectedDate)) : label}
+          >
+            <CalendarIcon aria-hidden={true} />
+          </DatePickerButton>
+        }
+        aria-autocomplete="none"
+        label={label}
+        {...props}
+      />
+
+      {inputRef && inputRef.current && inputRef.current.inputWrapperRef && (
```

Comment:
I wonder if we should add a condition on `visible` also to avoid useless rendering? 🤔